packages/php-wasm/web/vite.config.ts

```diff
@@ -90,18 +102,46 @@ export default defineConfig(({ command }) => {
 					if (
 						command === 'build' &&
 						typeof specifier === 'string' &&
-						specifier.match(/icudt74l\.js$/)
+						specifier.match(/icudt74l\.dat$/)
 					) {
 						/**
-						 * The ../ is weird but necessary to make the final build say
-						 * import("./shared/icudt74l.js")
+						 * The ../../../ is weird but necessary to make the final build say
+						 * import("./shared/icudt74l.dat")
 						 * and not
-						 * import("shared/icudt74l.js")
+						 * import("shared/icudt74l.dat")
 						 *
 						 * The slice(-2) will ensure the 'public/`
 						 * portion is removed.
 						 */
-						return '../' + specifier.split('/').slice(-2).join('/');
+						return (
+							'../../../' +
+							specifier.split('/').slice(-2).join('/')
+						);
+					}
+				},
+			},
+			{
+				name: 'preserve-extension-loaders-imports',
+
+				resolveDynamicImport(specifier): string | void {
+					if (
+						command === 'build' &&
+						typeof specifier === 'string' &&
+						specifier.match(/intl\.so$/)
+					) {
+						/**
+						 * The ../../../ is weird but necessary to make the final build say
+						 * import("./php/{mode}/extensions/intl/{php_version}/intl.so")
+						 * and not
+						 * import("php/{mode}/extensions/intl/{php_version}/intl.so")
+						 *
+						 * The slice(-6) will ensure the 'public/`
```

Comment:
there's no `public` in the example above, let's adjust it to match what's actually happening